File: springboot-backend/src/main/java/com/wedspot/backend/services/implementation/BookingService.java
package com.wedspot.backend.services.implementation;

import com.wedspot.backend.Model.Entity.Booking;
import com.wedspot.backend.repository.IBookingRepository;
import com.wedspot.backend.services.IBookingService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
@RequiredArgsConstructor
public class BookingService implements IBookingService {

    private final IBookingRepository bookingRepository;

    @Override
    public List<Booking> getAllBookings() {
        return bookingRepository.findAll();
    }

    @Override
    public List<Booking> getBookingsByClient(Long clientId) {
        return bookingRepository.findByClientId(clientId);
    }

    @Override
    public List<Booking> getBookingsByVendor(Long vendorId) {
        return bookingRepository.findByVendorId(vendorId);
    }

    @Override
    public Booking createBooking(Booking booking) {
        return bookingRepository.save(booking);
    }

    @Override
    public Optional<Booking> updateBookingStatus(Long id, String status) {
        return bookingRepository.findById(id)
                .map(booking -> {
                    booking.setStatus(status);
                    return bookingRepository.save(booking);
                });
    }
}

File: backend/Services/Implementations/BookingService.cs
using backend.Models.Entities;
using backend.Repositories;

namespace backend.Services
{
    public class BookingService : IBookingService
    {
        private readonly IBookingRepository _bookingRepository;

        public BookingService(IBookingRepository bookingRepository)
        {
            _bookingRepository = bookingRepository;
        }

        public async Task<List<Booking>> GetAllBookings()
        {
            return await _bookingRepository.GetAllAsync();
        }

        public async Task<List<Booking>> GetBookingsByClient(long clientId)
        {
            return await _bookingRepository.GetByClientIdAsync(clientId);
        }

        public async Task<List<Booking>> GetBookingsByVendor(long vendorId)
        {
            return await _bookingRepository.GetByVendorIdAsync(vendorId);
        }

        public async Task<Booking> CreateBooking(Booking booking)
        {
            await _bookingRepository.AddAsync(booking);
            return booking;
        }

        public async Task<Booking?> UpdateBookingStatus(long id, string status)
        {
            var booking = await _bookingRepository.GetByIdAsync(id);
            if (booking == null) return null;

            booking.Status = status;
            booking.UpdatedAt = DateTime.UtcNow;
            
            await _bookingRepository.UpdateAsync(booking);
            return booking;
        }
    }
}
